### DISCUSSION TITLE
Installing WSL image

### DISCUSSION THREAD

**Discussion by d0n13** [wixtoolset/issues#6781] [Questions]
Hi all
 
Is it possible to install WSL easily using wix. Things that need to be done:
 
1.	Activate Windows features if not already enabled and how to handle restart if enabling the feature requires it??
```
dism.exe /online /enable-feature /featurename:VirtualMachinePlatform /all /quiet /NoRestart
dism.exe /online /enable-feature /featurename:Microsoft-Windows-Subsystem-Linux /all /quiet
```
 
2.	Set the default WSL version to 2
`wsl --set-default-version 2`
 
3.	Install the WSL image 
`wsl --import {wsl-Image}`
 
4.	Set this image as the default
`wsl --set-default {wsl-Image}`
 
Ideally on uninstall it should do the following too:
 
```
wsl –terminate {wsl-Image}
wsl –unregister {wsl-Image}
```
 
Any thoughts on this?
Thanks
Donie

**Comment by robmen** [comment]
It'd definitely be possible to do with Burn. Not sure how easy it'd be.
